rules.py

```diff
@@ -410,12 +409,14 @@ def check(self, element):
             return
         valid_ocd_id = False
         referenced_gpunit = None
+        external_ids = []
         for gpunit in self.gpunits:
             if gpunit.get("objectId", None) == element.text:
                 referenced_gpunit = gpunit
-                for extern_id in gpunit.iter("ExternalIdentifier"):
+                external_ids = gpunit.findall(".//ExternalIdentifier")
+                for extern_id in external_ids:
                     id_type = extern_id.find("Type")
-                    if id_type is not None and id_type.text == "ocd-id":
+                    if id_type is not None and id_type.text.lower() == "ocd-id":
```

Comment:
Ok. Added a check for this. The schema rule should also catch it since the xsd requires it.